crates/circuit/src/interner.rs

```diff
@@ -50,17 +50,23 @@ unsafe impl<T: ?Sized> Sync for Interned<T> {}
 /// itself (the `Interned` type), rather than raw references; the `Interned` type is narrower than a
 /// true reference.
 ///
+/// This can only be implemented for owned types that implement `Default`, so that the convenience
+/// method `Interner::get_default` can work reliably and correctly; the "default" index needs to be
+/// guaranteed to be reserved and present for safety.
```

Comment:
The bit about safety might be better as an inline comment in the `new`, since it's more of an implementation detail, yeah?

Reply:
The same (or very similar) comment is on `new` as well - is there more you want to add to the `new` comment?

Reply:
Looking at this again, my issue was more that the text "This can only be implemented" makes it sound like the user would somehow be implementing the struct and would need to make this consideration (since it is in the docstring, rather than as a code comment). This is of course not possible, but if it were a trait, it would be.

A simple change to "This is only implemented" would make it read as less confusing, at least for me.

Reply:
Done in 547bd68389.